src/devices/machine/upd777_dev.h

```diff
@@ -0,0 +1,32 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+#ifndef MAME_MACHINE_UPD777
+#define MAME_MACHINE_UPD777
+
+#pragma once
+
+#include "emupal.h"
+#include "screen.h"
+
+#include "cpu/upd777/upd777.h"
+
+class upd777_device : public upd777_cpu_device {
+public:
+	upd777_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+protected:
+	virtual void device_add_mconfig(machine_config &config) override;
+	virtual const tiny_rom_entry *device_rom_region() const override;
+
+private:
+	uint32_t screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
+
+	void palette_init(palette_device &palette) const;
+
+	required_device<gfxdecode_device> m_gfxdecode;
+	required_device<palette_device> m_palette;
+};
+
+DECLARE_DEVICE_TYPE(UPD777, upd777_device)
```

Comment:
Unless MCU embedded peripherals exist as separate chips (eg. Toshiba Z80 with built-in CTC), they can be implemented in the CPU device.

Reply:
interesting, I was always told to keep video stuff out of the 'cpu' folder in the past

Reply:
Why? Only for video? I don't see how that makes sense while at the same time having eg. the versatile serial controller for Hitachi H8 or HD6301 in the CPU device.
Or, see the LCD controller in SM51x or the HT11xx that you recently added, that is video.

Reply:
well one of the previous rules was that things in CPU should be somewhat usable outside of MAME, with minimal effort, so using MAME specific video type stuff was considered bad.

If that has changed, fine.  I might still keep them in different files, for organization sake, but can put the _dev files in the CPU folder.

Reply:
I'm not aware of such a rule. Maybe you mean it was the opinion of someone that gave feedback on an older PR of yours.

Reply:
well given the amount of contradictory 'make it up as you go along' feedback I seem to get, that wouldn't surprise me.